File: Player.cpp
#include "Player.h"


void Player::setName(const std::string& n)
{
	name = n;
}
std::string Player::getName() const		//return name of player
{
	return name;
}
int Player::getMovesCount() const		//return number of moves
{
	return move_count;
}
void Player::incremetMove()	//increment number of moves
{
	move_count++;
}
std::string Player::convertToString() const	//convert the attributes of player object in a single string
{
	return name +" "+ std::to_string(move_count);
}


char Player::getInputChar() const
{
	return input_char;
}

File: web_based/backend/player.js
class Player {
    constructor(name, inputChar) {
        this.name = name;
        this.moves_count = 0;
        this.item = inputChar;
    }

    setName(n) {
        this.name = n;
    }

    getName() {
        return this.name;
    }

    getMoveCount() {
        return this.moves_count;
    }

    getInputChar() {
        return this.item;
    }

    incrementMoveCount() {
        this.moves_count++;
    }

    convertToString() {
        return `${this.name},${this.moves_count},${this.item}`;
    }
}
